File: Python-Solutions/IsomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        ds = {}
        dt = {}
        for i in range(len(s)):
            if s[i] in ds:
                if t[i] != ds[s[i]]:
                    return False
            elif t[i] in dt:
                if s[i] != dt[t[i]]:
                    return False
            else:
                ds[s[i]] = t[i]
                dt[t[i]] = s[i]
        return True

File: Swift-Solutions/IsomorphicStrings.swift
class Solution {
    func isIsomorphic(_ s: String, _ t: String) -> Bool {
        var map = [Character: Character]()
        for i in (0..<s.count){
            if map[s[i]] == nil{
                map[s[i]] = t[i]
            }
            else if map[s[i]] != t[i]{
                return false
            }
        }
        return true
    }
}

extension StringProtocol {
    subscript(offset: Int) -> Character {
        self[index(startIndex, offsetBy: offset)]
    }
}
